core/src/main/java/com/graphhopper/storage/index/QueryResult.java

```diff
@@ -49,8 +51,10 @@ public QueryResult(double queryLat, double queryLon) {
     }
 
     /**
-     * @return the closest matching node. -1 if nothing found, this should be avoided via a call of
-     * 'isValid'
+     * Returns the closest matching node. This is either a tower node of the base graph
+     * or a virtual node (see also {@link com.graphhopper.routing.QueryGraph#lookup(List)}).
```

Comment:
The developer shouldn't have to take care about this, but knows in general what the QueryGraph does (virt+real). We can keep it here but in general I would not want to add this separate meaning everywhere.